qcodes/dataset/measurements.py

```diff
@@ -404,40 +424,50 @@ def _validate_result_deps(
         try:
             self._interdeps.validate_subset(list(results_dict.keys()))
         except (DependencyError, InferenceError) as err:
-            raise ValueError('Can not add result, some required parameters '
-                             'are missing.') from err
+            raise ValueError(
+                "Can not add result, some required parameters are missing."
+            ) from err
 
     def _validate_result_shapes(
-            self, results_dict: Mapping[ParamSpecBase, values_type]) -> None:
+        self, results_dict: Mapping[ParamSpecBase, values_type]
+    ) -> None:
         """
         Validate that all sizes of the ``results_dict`` are consistent.
         This means that array-values of parameters and their setpoints are
         of the same size, whereas parameters with no setpoint relation to
         each other can have different sizes.
         """
-        toplevel_params = (set(self._interdeps.dependencies)
-                           .intersection(set(results_dict)))
+        toplevel_params = set(self._interdeps.dependencies).intersection(
```

Comment:
looks like the formatting of the whole file got changed - in this repo darker is used to make the formatting improvements gradual (by doing them only on the diffs and not the whole files) so that the formatting changes don't distract from the changes in the code. So, see if you can run darker (via pre commit hooks) instead of running black on the whole file.

Reply:
Yes. I have my IDE set up to format-on-save using Black. It's not clear to me how to roll back those formatting changes without doing it manually at this point. The other option that comes to mind is completely starting over. Both of these feel equally unappealing.

Reply:
If it helps, I believe this is a complete list of the methods which were changed beyond formatting:
DataSaver.add_result
DataSaver._conditionally_expand_parameter_with_setpoints
DataSaver._unpack_partial_result
Measurement.register_parameter
Measurement._register_parameter
Measurement._register_parameter_with_setpoints
str_or_register_name

Reply:
> It's not clear to me how to roll back those formatting changes without doing it manually at this point.

ok, then perhaps indeed too late. Perhaps going forward you could remove the automatic formatting via IDE and rely on pre-commit (which uses darker with black)?

Reply:
I also have my editor setup for auto formatting with black but in QCoDeS I have the following in my `.vscode\settings.json` file to avoid this taking effect on qcodes

```
"[python]": {
    "editor.defaultFormatter": "ms-python.black-formatter",
    "editor.formatOnType": true,
    "editor.formatOnSave": false
    },
...
```